### PR TITLE
feat: add channel broker implementation

### PR DESCRIPTION
**Pull request by HeerakKashyap**
## Description
This PR implements the channel broker functionality that was requested in meshsync issue #450.

**Related to:** meshsync issue #450 (https://github.com/meshery/meshsync/issues/450)

## Context
The original issue was filed in the meshsync repository, but the implementation needed to be moved to meshkit for better code organization and reusability. This PR addresses that requirement.

## Changes
- Add `broker/channel/` package with full implementation
- Implement `ChannelBrokerHandler` conforming to `broker.Handler` interface
- Add comprehensive unit tests covering all functionality including thread-safety
- Add proper error handling and timeout mechanisms
- Add configuration options for buffer size and publish delay

## Testing
- ✅ All unit tests pass
- ✅ Thread-safety tests pass
- ✅ Timeout scenarios tested
- ✅ Code compiles successfully

## Implementation Details
- Thread-safe implementation using `sync.RWMutex`
- Proper error handling in `PublishWithChannel` to prevent silent message loss
- Comprehensive test coverage including edge cases
- Follows existing meshkit patterns and conventions

Fixes #450 (https://github.com/meshery/meshsync/issues/450 - via meshsync repository)

### CODE REVIEW

**Comment by HeerakKashyap** [comment]
@leecalcote kindly review this.

File: broker/channel/channel.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+// PublishWithChannel - to publish messages with channel
+func (h *ChannelBrokerHandler) PublishWithChannel(subject string, msgch chan *broker.Message) error {
+	go func() {
+		// as soon as this channel will be closed, for loop will end
+		for msg := range msgch {
+			if err := h.Publish(subject, msg); err != nil {
+				// Log the error to help with debugging
+				// In a production environment, this should use a proper logger
+				fmt.Printf("Error publishing message to subject %s: %v\n", subject, err)
```
Comment (n2h9) [inline broker/channel/channel.go]: We actually have a proper logger, 
github.com/meshery/meshkit/logger -> Handler

you may wish to use it :innocent:

Comment (HeerakKashyap) [reply broker/channel/channel.go]: OKay Sir , I'll definitely give it a look @n2h9

**Comment by HeerakKashyap** [comment]
@n2h9 - I've updated the code to use the proper meshkit logger instead of fmt.Printf.

Changes made:
- Added logger.Handler field to ChannelBrokerHandler
- Added WithLogger option for custom logger injection  
- Now uses logger.Error() method for proper error logging
- Falls back to fmt.Printf only if logger creation fails

Thanks for the guidance!

File: broker/channel/channel.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+// PublishWithChannel - to publish messages with channel
+func (h *ChannelBrokerHandler) PublishWithChannel(subject string, msgch chan *broker.Message) error {
+	go func() {
+		// as soon as this channel will be closed, for loop will end
+		for msg := range msgch {
+			if err := h.Publish(subject, msg); err != nil {
+				// Use proper logger if available, otherwise fallback to fmt.Printf
+				if h.log != nil {
+					h.log.Error(err)
```
Comment (n2h9) [inline broker/channel/channel.go:143]: Probably makes sense to log here information about the  subject , similar to fmt.Printf version.

**Review by n2h9 — APPROVED** [review]
